core/src/main/java/org/infinispan/marshall/protostream/impl/SerializationContextRegistry.java

```diff
@@ -0,0 +1,33 @@
+package org.infinispan.marshall.protostream.impl;
+
+import org.infinispan.protostream.BaseMarshaller;
+import org.infinispan.protostream.FileDescriptorSource;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.SerializationContextInitializer;
+import org.infinispan.protostream.config.Configuration;
+
+/**
+ * Manages {@link SerializationContext} across modules for use by various components.
+ *
+ * @author Ryan Emerson
+ * @since 10.0
+ */
+public interface SerializationContextRegistry {
+
+   void addProtoFile(MarshallerType type, FileDescriptorSource fileDescriptorSource);
+
+   void addMarshaller(MarshallerType type, BaseMarshaller marshaller);
```

Comment:
Will this method ever be used outside of the `PERSISTENCE` type?

Reply:
It will definitely be required for the GlobalMarshaller work. At the very least it will be used to register the `MarshallableUserObject` marshaller.

Reply:
Yeah, I kept getting user marshaller confused vs persistence and cluster communication